### PR TITLE
Add test cases to the test suite "changeParams"

### PR DESCRIPTION
**Pull request by HoOngEe**

### CODE REVIEW

File: test/src/e2e.dynval/dv.changeParams.test.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+import { SDK } from "codechain-sdk";
+import * as stake from "codechain-stakeholder-sdk";
+import "mocha";
+
+import { validators as originalValidators } from "../../tendermint.dynval/constants";
+import { PromiseExpect } from "../helper/promise";
+import { withNodes, changeParams, defaultParams } from "./setup";
+import { faucetSecret, faucetAddress } from "../helper/constants";
+
+async function getPossibleAuthors(
```
Comment (majecty) [inline test/src/e2e.dynval/dv.changeParams.test.ts]: Please use the stake's getPossibleAuthors.

File: test/src/e2e.dynval/dv.changeParams.test.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+            await checkingNode.waitForTx(changeTxHash);
+            await checkingNode.waitForTermChange(2, termSeconds * margin * 2);
+
+            const termMetadata1 = (await stake.getTermMetadata(
+                checkingNode.sdk
+            ))!;
+            const prevBlock1 = (await checkingNode.sdk.rpc.chain.getBlock(
+                termMetadata1.lastTermFinishedBlockNumber - 1
+            ))!;
```
Comment (majecty) [inline test/src/e2e.dynval/dv.changeParams.test.ts]: Could you use a more descriptive variable name? It is hard to understand the prevBlock is whose prev block.

**Comment by HoOngEe** [comment]
I have implemented 3 test cases for the suite `changeParams`. I found an election error in the case that I changed the parameter `min_num_of_validators`. It was expected that 6 candidates would have been elected as validators but there were 7 validators in the next term.

File: test/src/e2e.dynval/dv.changeParams.test.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        it("Term seconds should be changed", async function() {
+            const margin = 1.2;
+            const checkingNode = nodes[0];
+            const termSeconds = 10;
+            const overrideParams = {
+                termSeconds
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
+                ...defaultParams,
+                ...overrideParams
```
Comment (foriequal0) [inline test/src/e2e.dynval/dv.changeParams.test.ts]: You can write like this to reduce indirection.
```ts
const changeTxHash = await changeParams(checkingNode, 1, {
    ...defaultParams,
    termSeconds: 10,
```
And with this code, you could have overwritten some parameters. You should do instead:
```ts
const initialParams = {
    ...defualtParams,
    termSeconds: 20,
    maxNumOfValidators: 8
};
const node = withNodes(..., { overrideParams: initialParams });

it(..., async function() {
    const changeTxHash = await changeParams(checkingNode, 1, {
        ...initialParams,
        termSeconds: 10
    });
});
```

File: test/src/e2e.dynval/dv.changeParams.test.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        });
+    });
+
+    describe("Change minimum fee", async function() {
+        it("Change minimum fee of pay transaction", async function() {
+            const checkingNode = nodes[0];
+            const overrideParams = {
+                minPayCost: 11
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
```
Comment (foriequal0) [inline test/src/e2e.dynval/dv.changeParams.test.ts]: ditto

File: test/src/e2e.dynval/dv.changeParams.test.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+            const termSeconds = 40;
+            const margin = 1.2;
+            const [alice, betty, charlie, dorothy, ...left] = allDynValidators;
+            const checkingNode = nodes[0];
+            const overrideParams = {
+                minNumOfValidators: 6,
+                termSeconds
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
```
Comment (foriequal0) [inline test/src/e2e.dynval/dv.changeParams.test.ts]: ditto

File: test/src/e2e.dynval/dv.changeParams.test.ts (hunk 1 above)
Comment (HoOngEe) [reply test/src/e2e.dynval/dv.changeParams.test.ts]: Changed

File: test/src/e2e.dynval/dv.changeParams.test.ts (hunk 2 above)
Comment (HoOngEe) [reply test/src/e2e.dynval/dv.changeParams.test.ts]: I changed the variable names.

File: test/src/e2e.dynval/dv.changeParams.test.ts (hunk 3 above)
Comment (HoOngEe) [reply test/src/e2e.dynval/dv.changeParams.test.ts]: I applied it.

File: test/src/e2e.dynval/dv.changeParams.test.ts [hunk 6]
```diff
@@ -134,4 +135,71 @@ describe("Change commonParams", function() {
...
+            sdk,
+            currentTermInitialBlockNumber
+        ))!.map(platformAddr => platformAddr.toString());
+
+        expect(termMetadata.currentTermId).to.be.equals(2);
+        expect(validatorsAfter.length).to.be.equals(target.length);
+        expect(validatorsAfter).contains.all.members(target);
+    }
+
+    describe.only("Change the minimum number of validators", async function() {
```
Comment (Remagpie) [inline test/src/e2e.dynval/dv.changeParams.test.ts]: Remove `only`

**Pull request merged**